Index: src/com/badlogic/gdx/backends/android/AndroidApplication.java
===================================================================
--- src/com/badlogic/gdx/backends/android/AndroidApplication.java	(revision 2978)
+++ src/com/badlogic/gdx/backends/android/AndroidApplication.java	(working copy)
@@ -185,15 +185,18 @@
 	@Override
 	protected void onPause () {
 		if (wakeLock != null) wakeLock.release();
-		graphics.pause();
+		if(graphics!=null)graphics.pause();
 
+		if(input != null){
+			
 		input.unregisterSensorListeners();
 		// erase pointer ids. this sucks donkeyballs...
 		int[] realId = input.realId;
 		for (int i = 0; i < realId.length; i++)
 			realId[i] = -1;
-
-		if (isFinishing()) {
+		}
+		
+		if (isFinishing() && graphics!=null) {
 			graphics.clearManagedCaches();
 			graphics.destroy();
 		}
@@ -208,6 +211,10 @@
 
 	@Override
 	protected void onResume () {
+		if(!graphics.isShown){
+			super.onResume();
+			return;
+		}
 		if (wakeLock != null) wakeLock.acquire();
 		Gdx.app = this;
 		Gdx.input = this.getInput();
@@ -215,7 +222,8 @@
 		Gdx.files = this.getFiles();
 		Gdx.graphics = this.getGraphics();
 
-		((AndroidInput)getInput()).registerSensorListeners();
+		if(input != null)((AndroidInput)getInput()).registerSensorListeners();
+		
 		if (audio != null) audio.resume();
 
 		if (graphics != null && graphics.view != null) {
Index: src/com/badlogic/gdx/backends/android/AndroidGraphics.java
===================================================================
--- src/com/badlogic/gdx/backends/android/AndroidGraphics.java	(revision 2978)
+++ src/com/badlogic/gdx/backends/android/AndroidGraphics.java	(working copy)
@@ -361,11 +361,15 @@
 		}
 	}
 
+	public boolean isShown = true;
+	
 	void destroy () {
 		synchronized (synch) {
 			running = false;
 			destroy = true;
 
+			if(!isShown)return;
+			
 			while (destroy) {
 				try {
 					synch.wait();
